Framework/Algorithms/src/XDataConverter.cpp

```diff
@@ -61,6 +65,12 @@ void XDataConverter::exec() {
   if (inputWS->axes() > 1)
     outputWS->replaceAxis(1, std::unique_ptr<API::Axis>(inputWS->getAxis(1)->clone(outputWS.get())));
 
+  bool isRebinnedWorkspace = inputWS->id() == "RebinnedOutput";
+  RebinnedOutput_sptr outRB = std::dynamic_pointer_cast<RebinnedOutput>(outputWS);
+  // Force fractions to unity (converting from histo to point discards bin info).
+  MantidVecPtr outF;
+  outF.access().resize(inputWS->getNumberBins(0), 1.0);
```

Comment:
Couple of things relating to these few lines:
- the number of bins could in theory be different on each spectra so please can you move the declaration\initialisation of outF into the loop so you can call getNumberBins(i)? I appreciate this doesn't happen v often but since this is a very commonly used algorithm it's worth doing I think
- Please can you check outRB isn't empty before it's used? Just add `if (outRB)` before it's used below. We run Coverity scans on the code outside of Jenkins and it will flag this up

Reply:
for the second point I assume you mean to check its not None?